packages/asset-swapper/src/utils/order_prune_utils.ts

```diff
@@ -0,0 +1,85 @@
+    constructor(devUtils: DevUtilsContract, opts: OrderPrunerOpts ) {
+        this.expiryBufferMs = opts.expiryBufferMs;
+        this._devUtils = devUtils;
+    }
+
+    public async pruneSignedOrdersAsync(signedOrders: SignedOrder[]): Promise<PrunedSignedOrder[]> {
+        const unsortedOrders = this._filterForUsableOrders(
+            signedOrders,
+            this.expiryBufferMs / constants.ONE_SECOND_MS,
+        );
+
+        const signatures = _.map(unsortedOrders, o => o.signature);
+        const [ordersInfo, remainingTakerAssetAmounts, isValidSignatures ] = await this._devUtils.getOrderRelevantStates.callAsync(
+            unsortedOrders,
+            signatures,
+        );
+        const ordersOnChainMetadata: any[] = ordersInfo.map((orderInfo, index) => {
+            return {
+                ...orderInfo,
+                remainingTakerAssetAmount: remainingTakerAssetAmounts[index],
+                isValidSignature: isValidSignatures[index],
+            };
+        });
+        // take orders + on chain information and find the valid orders and remaining fillable maker asset amounts
+        const prunedOrders = this._filterForFillableAndValidOrders(unsortedOrders, ordersOnChainMetadata);
+
+        return prunedOrders;
+    }
+
+    // tslint:disable-next-line: prefer-function-over-method
+    private _filterForFillableAndValidOrders(
+        orders: SignedOrder[],
+        ordersOnChainMetadata: any[],
+    ): PrunedSignedOrder[] {
+        const result = _.chain(orders).filter((order: SignedOrder, index: number): boolean => {
+            const { isValidSignature, orderStatus, remainingTakerAssetAmount } = ordersOnChainMetadata[index];
+            return isValidSignature &&
+            orderStatus === OrderStatus.Fillable &&
+            remainingTakerAssetAmount.gt(constants.ZERO_AMOUNT)
+            && utils.isOrderTakerFeePayableWithMakerAsset(order);
+        }).map((order: SignedOrder, index: number): PrunedSignedOrder => {
+            const { remainingTakerAssetAmount } = ordersOnChainMetadata[index];
+            return {
+                ...order,
+                remainingTakerAssetAmount,
+                remainingMakerAssetAmount: orderCalculationUtils.getMakerFillAmount(order, remainingTakerAssetAmount),
+                remainingTakerFee: orderCalculationUtils.getTakerFeeAmount(order, remainingTakerAssetAmount),
+            };
+        }).value();
+        return result;
+    }
+
+    // tslint:disable-next-line: prefer-function-over-method
+    private _filterForUsableOrders(
+        orders: SignedOrder[],
+        expiryBufferMs: number,
+    ): SignedOrder[] {
+        const result = _.filter(orders, order => {
+            return (
+                orderCalculationUtils.isOpenOrder(order) &&
+                utils.isOrderTakerFeePayableWithMakerAsset(order) &&
+                !orderCalculationUtils.willOrderExpire(order, expiryBufferMs / constants.ONE_SECOND_MS)
```

Comment:
I believe `this.expiryBufferMs / constants.ONE_SECOND_MS` has already been applied in the caller. What unit should this parameter be in when passed into the utils `this.expiryBufferMs` already reads as milliseconds.